### PR TITLE
More Q* subclasses improvements

### PR DESCRIPTION
**Pull request by zas**
# Summary

* This is a…
  * [ ] Bug fix
  * [ ] Feature addition
  * [x] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-XXX

# Solution

# Action

Additional actions required:
* [ ] Update Picard [documentation](https://github.com/metabrainz/picard-docs) (please include a reference to this PR)
* [ ] Other (please specify below)

### CODE REVIEW

File: picard/ui/__init__.py [hunk 1]
```diff
@@ -212,27 +212,22 @@ def show_help(self):
...
-        self.id = uuid.uuid4()
+class HashableTreeWidgetItem(HashableItem, QtWidgets.QTreeWidgetItem):
+    pass
 
-    def __eq__(self, other):
-        return self.id == other.id
 
-    def __hash__(self):
-        return hash(str(self.id))
+class HashableListWidgetItem(HashableItem, QtWidgets.QListWidgetItem):
```
Comment (phw) [inline picard/ui/__init__.py:232]: I'd turn this around to `class HashableListWidgetItem(QtWidgets.QListWidgetItem, HashableItem)`.

The reason is that I would first list the primary type we try to inherit. `HashableListWidgetItem` first and foremost is supposed to be a `QListWidgetItem`. The `HashableItem` class acts more as a mixin that ads extra functionality.

Same for the `HashableTreeWidgetItem` above.

Comment (zas) [reply picard/ui/__init__.py:232]: I tried, but when doing so, I get an error:

```
Traceback (most recent call last):
  File "/home/zas/src/picard/./picard/ui/mainwindow/__init__.py", line 905, in show_options
    options_dialog = OptionsDialog.show_instance(page, self)
  File "/home/zas/src/picard/./picard/ui/__init__.py", line 156, in show_instance
    instance = cls.get_instance(*args, **kwargs)
  File "/home/zas/src/picard/./picard/ui/__init__.py", line 150, in get_instance
    cls._instance = cls(*args, **kwargs)
  File "/home/zas/src/picard/./picard/ui/options/dialog.py", line 230, in __init__
    self.add_pages(None, default_pagename, self.ui.pages_tree)
  File "/home/zas/src/picard/./picard/ui/options/dialog.py", line 164, in add_pages
    self.item_to_page[item] = page
TypeError: unhashable type: 'HashableTreeWidgetItem'
```

Comment (phw) [reply picard/ui/__init__.py:232]: Ok, the wonders of multiple inheritance :shrug:

**Review by phw — APPROVED** [review]

File: picard/ui/__init__.py (hunk 1 above)
Comment (zas) [reply picard/ui/__init__.py:232]: Yes, they never really work as we expect... ;)

**Pull request merged**